main.tf

```diff
@@ -284,10 +296,18 @@ resource "ibm_container_vpc_worker_pool" "autoscaling_pool" {
 # push down an updated vpn config, and then the vpn server and client need
 # to pick up this updated config. Depending on how busy the network
 # microservice is handling requests, there might be a delay.
+
+# Please note - This network check is only applicable in case of public
+# endpoints and will not be checked if private endpoint is used.
+
 ##############################################################################
 
 resource "null_resource" "confirm_network_healthy" {
 
+  # If private endpoint then do not create this resource.
+  count = var.disable_public_endpoint ? 0 : 1
```

Comment:
hmm, didn't we say that the consumer may have access to their private endpoint, and so may still wish to run this script. Instead we should probably have a variable called `confirm_network_healthy` and default it to `true`, however in the variable description make sure it mentions that in order to set to true, the runtime must have access to kube